### PR TITLE
fix(pip.install.wheel): support variations on `pythonw.exe`

### PR DESCRIPTION
**Pull request by paveldikov**
Fixes: #5256

## Summary

Emulate `pip`'s behaviour and find `pythonw` executable by doing an `s/python/pythonw/g` style transformation, as opposed to assuming a constant `pythonw.exe` path.

See #5256 for more detail e.g. why this is a useful behaviour to emulate.

## Test Plan

Added this scenario to `test_script_executable`.

---
© 2024 Morgan Stanley.

THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE APACHE LICENSE V.2.0 AND THE MIT LICENSE.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

**Comment by zanieb** [comment]
Thanks for contributing! @charliermarsh will own review of this one.

File: crates/install-wheel-rs/src/wheel.rs [hunk 1]
```diff
@@ -237,8 +237,11 @@ fn get_script_executable(python_executable: &Path, is_gui: bool) -> PathBuf {
     // Only check for pythonw.exe on Windows
     if cfg!(windows) && is_gui {
         python_executable
-            .parent()
-            .map(|parent| parent.join("pythonw.exe"))
+            .file_name()
```
Comment (samypr100) [inline crates/install-wheel-rs/src/wheel.rs:240]: Might be useful to include the pip reference to [_get_alternate_executable](https://github.com/pypa/pip/blob/76e82a43f8fb04695e834810df64f2d9a2ff6020/src/pip/_vendor/distlib/scripts.py#L121-L126)

e.g.

 ```rust
/// Returns a [`PathBuf`] to `python[w].exe` for script execution.
///
/// <https://github.com/pypa/pip/blob/76e82a43f8fb04695e834810df64f2d9a2ff6020/src/pip/_vendor/distlib/scripts.py#L121-L126>
 ```

Comment (paveldikov) [reply crates/install-wheel-rs/src/wheel.rs:242]: Added.

**Comment by charliermarsh** [comment]
Do you mind confirming for me that your contribution is licensed under both the Apache _and_ MIT license? The project is dual-licensed so inbound contributions must be covered by both, and then consumers of the project can consume under the terms of either.

**Comment by paveldikov** [comment]
@charliermarsh -- that is OK (I have reviewed this internally and amended the PR header)

**Comment by charliermarsh** [comment]
@paveldikov -- Thank you, I will get this merged today.

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
Thanks!

**Pull request merged**